scipy/sparse/_coo.py

```diff
@@ -572,49 +569,62 @@ def eliminate_zeros(self):
     #######################
 
     def _add_dense(self, other):
-        if other.shape != self.shape:
-            raise ValueError(f'Incompatible shapes ({self.shape} and {other.shape})')
+        if self.shape != other.shape:
+            bshape = np.broadcast_shapes(self.shape, other.shape)
+            self = self._broadcast_to(bshape)
+            other = np.broadcast_to(other, bshape)
         dtype = upcast_char(self.dtype.char, other.dtype.char)
         result = np.array(other, dtype=dtype, copy=True)
         fortran = int(result.flags.f_contiguous)
-        if self.ndim == 1:
-            coo_todense_nd(np.array([1]), self.nnz, self.ndim,
-                           self.coords[0], self.data, result.ravel('A'), fortran)
-        elif self.ndim == 2:
+        if self.ndim < 3 and len(other.shape) < 3:
             M, N = self._shape_as_2d
             coo_todense(M, N, self.nnz, self.row, self.col, self.data,
                         result.ravel('A'), fortran)
         else:
-            if fortran:
-                strides = np.append(1, np.cumprod(self.shape[:-1]))
-            else:
-                strides = np.append(np.cumprod(self.shape[1:][::-1])[::-1], 1)
-            coords = np.concatenate(self.coords)
-            coo_todense_nd(strides, self.nnz, self.ndim,
-                           coords, self.data, result.ravel('A'), fortran)
+            result_shape = self.shape
+            self = self.reshape(1, -1)
+            other = other.reshape(1, -1)
+            result = _spbase._add_dense(self, other)
+            # reshape back to n-D
+            result = result.reshape(result_shape)
```

Comment:
This seems like it could be rather expensive, as reshaping a COO array requires reallocating, and the base _add_dense will convert to CSR format and allocate even more.

How about:

```python
self.sum_duplicates()
result[*self.coords] += self.data
```

Reply:
Done! (b619b2b7dc5dfed3daad7a92a5b92d384e54149e)

Reply:
Ah, the unpacking operator is giving syntax error for older python versions I think. Could I try replacing that with `result[tuple(self.coords)] += self.data`? Would that lead to copying of data or does `tuple()` just create an object referencing `self.coords`?

Reply:
`self.coords` is already a tuple of 1d arrays. 
So try `result[self.coords] += self.data`. 
That should work for both  Python 3.10 and 3.13